analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -1,5 +1,6 @@
 ﻿using System;
```

Comment:
Additional test cases which might be worth adding:
- `Dispose()` is called through the `?.` operator
- interface with method implementation (C# 8)
- using the `record struct` and `record class` types (C# 10)

Reply:
For `record struct` and `record class` are the tests really needed for this rule - since we are detecting invocations?

Reply:
IDisposable record struct and IDiposable record class? Sounds reasonable

Reply:
`d?.Dispose()` is a good idea to test, and assert message

Reply:
`d?.Dispose` it's a false negative - no message to assess.